### PR TITLE
Refine collection lock usage

### PR DESCRIPTION
**Pull request by czs007**
issue #20256 
Signed-off-by: zhenshan.cao <zhenshan.cao@zilliz.com>

### CODE REVIEW

File: internal/querynode/collection.go [hunk 1]
```diff
@@ -339,19 +246,29 @@ func (c *Collection) isPDeltaChannelExist(channel string) bool {
...
+	c.markDestroyed = true
 }
 
-// getReleaseTime gets the time when collection is released
-func (c *Collection) getReleaseTime() (Timestamp, bool) {
-	c.releaseMu.RLock()
-	defer c.releaseMu.RUnlock()
-	return c.releaseTime, c.released
+// isHealthy get whether the collection is healthy
+func (c *Collection) healthyCheck() bool {
```
Comment (bigsheeper) [inline internal/querynode/collection.go:260]: function name is mismatch with comment.

File: internal/querynode/meta_replica.go [hunk 2]
```diff
@@ -226,18 +226,14 @@ func (replica *metaReplica) removeCollectionPrivate(collectionID UniqueID) error
 	if err != nil {
 		return err
 	}
-
-	// block incoming search&query
-	collection.mu.Lock()
-	defer collection.mu.Unlock()
-
+	// blocking query & search
```
Comment (bigsheeper) [inline internal/querynode/meta_replica.go:229]: this comment would remove

File: internal/querynode/plan.go [hunk 3]
```diff
@@ -40,8 +40,10 @@ type SearchPlan struct {
 
 // createSearchPlan returns a new SearchPlan and error
 func createSearchPlan(col *Collection, dsl string) (*SearchPlan, error) {
```
Comment (bigsheeper) [inline internal/querynode/plan.go:42]: maybe could move to collection.go?

File: internal/querynode/collection.go [hunk 4]
```diff
@@ -339,19 +247,29 @@ func (c *Collection) isPDeltaChannelExist(channel string) bool {
...
 
-// setReleaseTime records when collection is released
-func (c *Collection) setReleaseTime(t Timestamp, released bool) {
-	c.releaseMu.Lock()
-	defer c.releaseMu.Unlock()
-	c.releaseTime = t
-	c.released = released
+// MarkDestroyed mark the collection to be/ has been destroyed
+func (c *Collection) MarkDestroyed() {
+	if c == nil {
```
Comment (jiaoew1991) [inline internal/querynode/collection.go:252]: In which case `c` will be nil

Comment (czs007) [reply internal/querynode/collection.go:252]: Just for prevention, hardly ever

**Pull request closed without merging**